lib/elixir/lib/file.ex

```diff
@@ -461,7 +461,7 @@ defmodule File do
   error will be returned.
   """
   @spec cp(Path.t, Path.t, (Path.t, Path.t -> boolean)) :: :ok | {:error, posix}
-  def cp(source, destination, callback \\ fn(_, _) -> true end) do
+  def cp(source, destination, callback \\ &path_differs?/2) do
```

Comment:
This means if the user is passing a custom function, they will need to add this logic himself. I don't think this is behaviour we want, right? We don't want to ever copy, we don't want to even invoke the callback. No?